driver-core/src/main/com/mongodb/connection/QueryResult.java

```diff
@@ -77,6 +96,19 @@ public ServerCursor getCursor() {
         return results;
     }
 
+    /**
+     * Gets the postBatchResumeToken.
+     *
+     * @return the postBatchResumeToken
+     * @since 3.11
+     */
+    public BsonDocument getPostBatchResumeToken() {
```

Comment:
Rather than adding a `getBatchResumeToken` method here consider adding a method to return the `cursorDocument`.  That should be more flexible for future and means the QueryResult class can be ignorant of the `POST_BATCH_RESUME_TOKEN`.

Reply:
I'm concerned about this change.  If we keep the cursorDocument around we've effectively doubled our memory footprint for query results: one copy for the `BsonDocument cursorDocument` and one for the `List<T> results`.  I think we have to find another way to abstract this.

Reply:
I think we either leak the resume token API into all cursors, or we have to do more surgery and add the cursor document and the results extractor method into QueryResult.

Reply:
I have removed this method from `QueryResult` and set the `postBatchResumeToken` in `QueryBatchCursor#initFromCommandResult`.